src/c_kzg_4844.c

```diff
@@ -1049,55 +1050,60 @@ static C_KZG_RET compute_kzg_proof_impl(
 /**
  * Compute KZG proof for polynomial in Lagrange form at position z.
  *
- * @param[out] out  The combined proof as a single G1 element
- * @param[in]  blob The blob (polynomial) to generate a proof for
- * @param[in]  z    The generator z-value for the evaluation points
- * @param[in]  s    The settings containing the secrets, previously
- *                  initialised with #new_kzg_settings
+ * @param[out] proof_out The combined proof as a single G1 element
+ * @param[out] y_out     The evaluation of the polynomial at the evaluation point z
+ * @param[in]  blob      The blob (polynomial) to generate a proof for
+ * @param[in]  z         The generator z-value for the evaluation points
+ * @param[in]  s         The settings containing the secrets, previously
+ *                       initialised with #new_kzg_settings
  */
 C_KZG_RET compute_kzg_proof(
-    KZGProof *out,
+    KZGProof *proof_out,
+    Bytes32 *y_out,
     const Blob *blob,
     const Bytes32 *z_bytes,
     const KZGSettings *s
 ) {
     C_KZG_RET ret;
     Polynomial polynomial;
-    fr_t frz;
+    fr_t frz, fry;
 
     ret = blob_to_polynomial(&polynomial, blob);
     if (ret != C_KZG_OK) goto out;
     ret = bytes_to_bls_field(&frz, z_bytes);
     if (ret != C_KZG_OK) goto out;
-    ret = compute_kzg_proof_impl(out, &polynomial, &frz, s);
+    ret = compute_kzg_proof_impl(proof_out, &fry, &polynomial, &frz, s);
     if (ret != C_KZG_OK) goto out;
+    bytes_from_bls_field(y_out, &fry);
 
 out:
     return ret;
 }
 
 /**
  * Helper function for compute_kzg_proof() and
- * compute_aggregate_kzg_proof().
+ * compute_blob_kzg_proof().
  *
- * @param[out] out        The combined proof as a single G1 element
+ * @param[out] proof_out  The combined proof as a single G1 element
+ * @param[out] y_out      The evaluation of the polynomial at the evaluation point z
  * @param[in]  polynomial The polynomial in Lagrange form
  * @param[in]  z          The evaluation point
  * @param[in]  s          The settings containing the secrets, previously
  *                        initialised with #new_kzg_settings
  */
 static C_KZG_RET compute_kzg_proof_impl(
-    KZGProof *out,
+    KZGProof *proof_out,
+    fr_t *y_out,
     const Polynomial *polynomial,
     const fr_t *z,
     const KZGSettings *s
 ) {
     C_KZG_RET ret;
-    fr_t y;
+    fr_t *y = y_out;
```

Comment:
Hmm, this seems a bit non-standard. Why do we need both `y` and `y_out`?

If we need both, I think it's more standard to use `y` throughout the function and then in the end do `*y_out = y`.

Reply:
I agree this a little weird. You could also delete the `y` variable and use `y_out` throughout the function.

Reply:
I just thought `y_out` is too long. However, I'm fine with either way. Note that `*y_out = y` may not work. I thought about it previously, but it will not work if `fr_t` is not copyable, so I avoided using it.

Reply:
> I just thought `y_out` is too long. However, I'm fine with either way. Note that `*y_out = y` may not work. I thought about it previously, but it will not work if `fr_t` is not copyable, so I avoided using it.

Hmm. That's true, but we've been assuming that `fr_t` is copyable in a bunch of places already (check for places where we are copying `FR_ZERO`).